packages/build/src/plugins/child/mutations.js

```diff
@@ -11,33 +11,34 @@ const { EVENTS } = require('../events')
 
 // `netlifyConfig` is read-only except for specific properties.
 // This requires a `Proxy` to:
-//  - Warn plugin authors when mutating read-only properties
-//  - Execute custom logic, e.g. setting normalized properties when denormalized
-//    ones are being modified
-const preventConfigMutations = function (netlifyConfig, configMutations, event) {
-  const state = { ongoingMutation: false }
```

Comment:
`state.ongoingMutation` is not needed now that side effects "handlers" are performed in a separate later step. This allows removing some of the logic below.